bson/binary.py

```diff
@@ -425,9 +435,19 @@ def from_vector(
         ...
 
     @classmethod
+    @overload
     def from_vector(
         cls: Type[Binary],
-        vector: Union[BinaryVector, list[int], list[float]],
+        vector: npt.NDArray[np.number],
+        dtype: BinaryVectorDtype,
+        padding: int = 0,
+    ) -> Binary:
+        ...
+
+    @classmethod
+    def from_vector(
+        cls: Type[Binary],
+        vector: Union[BinaryVector, list[int], list[float], npt.NDArray[np.number]],
```

Comment:
Is `BinaryVector` a subtype of list?

Reply:
No. It is more of a struct containing data (1d array/list of numbers) and metadata (dtype and padding).